project/src/Agents/GSTKinematicBody2DAgent.gd

```diff
@@ -15,30 +15,31 @@ var body: KinematicBody2D setget _set_body
 # SLIDE uses use move_and_slide
 # COLLIDE uses move_and_collide
 # POSITION changes the global_position directly
-var kinematic_movement_type: int = KinematicMovementType.SLIDE
+var movement_type: int
 
 var _last_position: Vector2
 
 
-func _init(body: KinematicBody2D) -> void:
+func _init(body: KinematicBody2D, movement_type: int = MovementType.SLIDE) -> void:
+	yield(body, "ready")
```

Comment:
If the body is _already_ ready, this yield never hits. It should check for if it's already in the tree first, though I'm fine with yielding instead of connecting.